charm-slurmctld/src/charm.py

```diff
@@ -416,6 +437,26 @@ def _resume_nodes(self, nodelist):
         update_cmd = f"update nodename={nodes} state=resume"
         self._slurm_manager.slurm_cmd('scontrol', update_cmd)
 
+    def _on_grafana_available(self, event):
+        """Create the grafana-source if we are the leader and have influxdb."""
+        leader = self._is_leader()
+        influxdb_info = self._get_influxdb_info()
+        grafana = self._grafana
+
+        if leader and influxdb_info:
+            grafana.set_grafana_source_info(influxdb_info)
+        else:
+            logger.debug("## on_grafana_available, but deferring")
+            event.defer()
```

Comment:
we might not need this here?

Reply:
I don't like this.

Reply:
Possibly we just log and drop the `defer()`?

Reply:
Should this log be a Warning or Error?

Reply:
Warning